third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -1573,8 +1573,8 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
             }
             let access_specifiers = et.translate_access_specifiers(&def.access_specifiers);
             let result = et.translate_seq(&loc, seq, &result_type, &ErrorMessageContext::Return);
-            et.finalize_types();
             let translated = et.post_process_body(result.into_exp());
+            et.finalize_types();
```

Comment:
This is not looking right. Finalization has to happen before post processing since this needs to have the ground types, it is not place by accident in this order. Its also happening inside of post process for some special cases [like here](https://github.com/aptos-labs/aptos-core/blob/a2db17a9fac269c0a92b28d6cc6fe56433628be3/third_party/move/move-model/src/builder/exp_builder.rs#L2233). Please look closer what is wrong here to find the right fix.

Reply:
The issue is that `finalize_types` generated the error before running `post_process_body`. I updated the code so that when `finalize_types` is run before `post_process_body`, error will not be generated.